tests/unscheduled/eip7939_count_leading_zeros/test_count_leading_zeros.py

```diff
@@ -0,0 +1,174 @@
+)
+def test_clz_opcode_scenarios(
+    state_test: StateTestFiller,
+    pre: Alloc,
+    test_id: str,
+    value: int,
+    expected_clz: int,
+):
+    """
+    Test CLZ opcode functionality.
+
+    Cases:
+    - Format 0xb000...111: leading zeros followed by ones (2**256 - 1 >> bits)
+    - Format 0xb010...000: single bit set at position (1 << bits)
+
+    Test coverage:
+    - Leading zeros pattern: 0b000...111 (0 to 256 leading zeros)
+    - Single bit pattern: 0b010...000 (bit at each possible position)
+    - Edge cases: CLZ(0) = 256, CLZ(2^256-1) = 0
+    """
+    sender = pre.fund_eoa()
+    contract_address = pre.deploy_contract(
+        code=Op.SSTORE(0, Op.CLZ(value)),
+        storage={"0x00": "0xdeadbeef"},
+    )
+    tx = Transaction(
+        to=contract_address,
+        sender=sender,
+        gas_limit=200_000,
+    )
+    post = {
+        contract_address: Account(storage={"0x00": expected_clz}),
+    }
+    state_test(pre=pre, post=post, tx=tx)
+
+
+@pytest.mark.valid_from("Osaka")
+def test_clz_gas(state_test: StateTestFiller, pre: Alloc, fork: Fork):
+    """Test CLZ opcode gas cost."""
+    contract_address = pre.deploy_contract(
+        Op.SSTORE(
+            0,
+            CodeGasMeasure(code=Op.CLZ(Op.PUSH1(1)), extra_stack_items=1, overhead_cost=0),
+        ),
+        storage={"0x00": "0xdeadbeef"},
+    )
+    sender = pre.fund_eoa()
+    tx = Transaction(to=contract_address, sender=sender, gas_limit=200_000)
+    post = {
+        contract_address: Account(  # Cost measured is CLZ + PUSH1
+            storage={"0x00": Spec.CLZ_GAS_COST + fork.gas_costs().G_VERY_LOW}
+        ),
+    }
+    state_test(pre=pre, post=post, tx=tx)
+
+
+@pytest.mark.valid_from("Osaka")
+def test_clz_stack_underflow(state_test: StateTestFiller, pre: Alloc):
+    """Test CLZ opcode with empty stack (should revert due to stack underflow)."""
+    sender = pre.fund_eoa()
+    contract_address = pre.deploy_contract(
+        code=Op.CLZ,
+        storage={"0x00": "0xdeadbeef"},
+    )
+    tx = Transaction(
+        to=contract_address,
+        sender=sender,
+        gas_limit=200_000,
+    )
+    post = {
+        contract_address: Account(
+            storage={"0x00": "0xdeadbeef"}  # Storage unchanged due to revert
+        ),
+    }
+    state_test(pre=pre, post=post, tx=tx)
+
+
+@pytest.mark.valid_at_transition_to("Osaka", subsequent_forks=True)
+def test_clz_fork_transition(blockchain_test: BlockchainTestFiller, pre: Alloc):
+    """Test CLZ opcode behavior at fork transition."""
+    sender = pre.fund_eoa()
+    code = Op.SSTORE(0, Op.CLZ(1 << 100))
+    contract_before = pre.deploy_contract(code)
+    contract_after = pre.deploy_contract(code)
+    blockchain_test(
+        pre=pre,
+        blocks=[
+            Block(
+                timestamp=14_999,
+                txs=[Transaction(to=contract_before, sender=sender, gas_limit=200_000)],
```

Comment:
This transaction should also revert right? Since CLZ is INVALID here so it OOGs?